LibBlosc2/src/decode.jl

```diff
@@ -0,0 +1,114 @@
+"""
+    Blosc2DecodingError()
+
+Error for data that cannot be decoded.
+"""
+struct Blosc2DecodingError <: DecodingError
+end
+
+function Base.showerror(io::IO, err::Blosc2DecodingError)
+    print(io, "Blosc2DecodingError: blosc2 compressed buffer cannot be decoded")
+    return nothing
+end
+
+"""
+    struct Blosc2DecodeOptions <: DecodeOptions
+    Blosc2DecodeOptions(; kwargs...)
+
+Blosc2 decompression using c-blosc2 library: https://github.com/Blosc/c-blosc2
+
+# Keyword Arguments
+
+- `codec::Blosc2Codec=Blosc2Codec()`
+
+# Keyword Arguments
+
+- `codec::Blosc2Codec=Blosc2Codec()`
+- `nthreads::Integer=1`: The number of threads to use
+"""
+struct Blosc2DecodeOptions <: DecodeOptions
+    codec::Blosc2Codec
+
+    nthreads::Int
+end
+function Blosc2DecodeOptions(; codec::Blosc2Codec=Blosc2Codec(),
+                             nthreads::Integer=1,
+                             kwargs...)
+    _nthreads = nthreads
+    check_in_range(1:typemax(Int32); nthreads=_nthreads)
+
+    return Blosc2DecodeOptions(codec, _nthreads)
+end
+
+# This decoder is thread safe: We don't use any of Blosc2's global variables.
+is_thread_safe(::Blosc2DecodeOptions) = true
```

Comment:
This should be removed because the blosc2 library is not written in a thread-safe way. For example, https://github.com/Blosc/c-blosc2/blob/40783eb33269883aa090daf04e091dbe39317fb9/blosc/schunk.c#L138 uses the `getenv` function which is not thread-safe.